### PR TITLE
ASCII .stl support

### PR DESCRIPTION
**Pull request by roystgnr**
@maxnezdyur reported we've had users ask for STL support, and I had some time today while waiting for long tests to finish running, so here's a first crack at it.

It's only ASCII and only Tri3 support currently, and I'm done for the week, but adding binary support and Tri6/7 subdivision-write support shouldn't be much extra work.

### CODE REVIEW

File: src/mesh/stl_io.C [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+
+
+void STLIO::read (const std::string & filename)
+{
+  LOG_SCOPE("read()", "STLIO");
+
+#ifndef LIBMESH_HAVE_CXX11_REGEX
+  libmesh_not_implemented();  // What is your compiler?!?  Email us!
+  libmesh_ignore(filename);
+#else
```
Comment (jwpeterson) [inline src/mesh/stl_io.C:154]: Should we make C++11 regex support officially required now? Looking at configure.ac, I see:
```
# --------------------------------------------------------------
# Test for optional modern C++ features, which libMesh offers shims
# for and/or which libMesh applications may want to selectively use.
# --------------------------------------------------------------
LIBMESH_TEST_CXX14_MAKE_UNIQUE
LIBMESH_TEST_CXX11_REGEX
```
and we obviously use `std::make_unique` freely everywhere now, so I think these could probably both be moved to the `ACSM_TEST_CXX_ALL` test, which is what sets the `have_cxx_all` flag. A topic for another PR though, either way.

File: src/mesh/stl_io.C [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+          libmesh_error_msg_if
+            (c != '\n' && c != '\r' &&
+             (c < ' ' || c > '~'),
+             "Found non-ASCII character " << int(c) << " on line " <<
+             line_num << "\nSTLIO does not yet support binary files.");
+        }
+
+      if (std::regex_search(line, sm, start_regex))
+        {
+          if (std::regex_search(line, sm, start_with_name_regex))
```
Comment (jwpeterson) [inline src/mesh/stl_io.C:253]: Our current `std::regex_search` configure test does not test the version of that [function](https://en.cppreference.com/w/cpp/regex/regex_search) taking a `std::match_results<T>`, so if we do decide to make `std::regex` required, we should probably also beef up the test a little, so it at least tests all the functionality we use.

**Review by jwpeterson — APPROVED** [review]

File: src/mesh/stl_io.C (hunk 1 above)
Comment (roystgnr) [reply src/mesh/stl_io.C:154]: Yeah, that sounds good to me.

[cppreference claims](https://en.cppreference.com/w/cpp/compiler_support/11) that Embarcadero C++ Builder and Sun/Oracle C++ are the only libraries that didn't add regex support ... but [later claims](

File: src/mesh/stl_io.C (hunk 2 above)
Comment (roystgnr) [reply src/mesh/stl_io.C:253]: Oh, yikes, definitely.  I'm already using `match_results` in our new `SparseMatrix::read_matlab` too.

**Pull request merged**